src/DecafVersionChecker.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+const style = `
+.version-modal {
+  background-color: rgba(0, 0, 0, 0.5);
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  z-index: 9999;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-size: 1.4em;
+}
+
+.version-modal-title {
+  text-align: center;
+  padding: 10px 20px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.version-modal-title h2 {
+  font-size: 1.5rem;
+  font-weight: bold;
+  color: white;
+  margin: 0;
+  margin-left: 10px;
+}
+
+.version-modal .version-modal-body {
+  display: flex;
+  flex-direction: column;
+  width: 50%;
+  background-color: #333;
+  border-radius: 5px;
+}
+
+.version-modal .version-modal-content {
+  padding: 5px 20px;
+}
+
+.version-modal .version-modal-footer {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  width: 100%;
+  padding: 10px;
+  margin-top: 20px;
+}
+.version-modal .reload-btn {
+  background-color: #177ddc;
+  color: white;
+  border: none;
+  width: 150px;
+  padding: 10px;
+  cursor: pointer;
+}
+
+.version-modal .reload-btn:hover {
+  background-color: #095cb5;
+}
+
+.version-modal .cancel-btn {
+  background-color: #333;
+  color: white;
+  border: none;
+  padding: 10px;
+  cursor: pointer;
+}
+
+.version-modal .cancel-btn:hover {
+  background-color: #444;
+}
+`;
+
+export interface DecafVersionCheckerProps {
+  currentVersion: string;
+  onNewVersion?: (newVersion: string) => void;
+}
+
+export default function DecafVersionChecker(props: DecafVersionCheckerProps) {
+  const [newVersion, setNewVersion] = useState();
+  const interval = useRef<number>();
+
+  useEffect(() => {
+    if (interval.current) {
+      clearInterval(interval.current);
+    }
+    interval.current = window.setInterval(() => {
+      fetch('/version.json')
```

Comment:
Can you append a querystring, something like `?random=${Math.random()}` to avoid cache (just in case)?

Reply:
Hmm... Does it go to:

1. `https://hebele.hubele.com/version.json`, or
2. `https://hebele.hubele.com/webapps/zartzurt/production/version.json`?